src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js

```diff
@@ -35,11 +35,37 @@ PrimeFaces.widget.MenuButton = PrimeFaces.widget.BaseWidget.extend({
         this.cfg.disabled = this.button.is(':disabled');
 
         if(!this.cfg.disabled) {
-            this.bindEvents();
+            this.bindButtonEvents();
             PrimeFaces.utils.registerDynamicOverlay(this, this.menu, this.id + '_menu');
         }
     },
 
+    /**
+     * @override
+     * @inheritdoc
+     * @param {JQuery} menuitem 
+     * @param {JQuery} submenu 
+     */
+    showSubmenu: function(menuitem, submenu) {
+        var pos = {
+            my: 'left top',
+            at: 'right top',
+            of: menuitem,
+            collision: 'flipfit'
+        };
+
+        //avoid queuing multiple runs
+        if(this.timeoutId) {
+            clearTimeout(this.timeoutId);
+        }
+
+        this.timeoutId = setTimeout(function () {
+           submenu.css('z-index', PrimeFaces.nextZindex())
+                  .show()
+                  .position(pos)
+        }, this.cfg.delay);
+    },
+
     /**
      * @override
```

Comment:
You can remove the @override and @inheritdoc here as now that you have renamed the method its no longer overriding the parent.

Reply:
`bindButtonEvents` method does not have `@override` and `@inheritdoc`. This line is for `refresh` method.

Reply:
sorry the way the review looks the code folding confused me. I see it in the raw file now.